[CI] KeyRotationIT testInitialInstallProducesSingleActiveKeyWithEmptyHandlerProgress failing

**Build Scans:**
- [elasticsearch-intake #40936 / part2](https://gradle-enterprise.elastic.co/s/whgchcqglq2pq)
- [elasticsearch-periodic-java-ea #289 / release-tests](https://gradle-enterprise.elastic.co/s/g5glrwgclzcgi)
- [elasticsearch-pull-request #152145 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/xxztvf7v2mwiw)
- [elasticsearch-pull-request #152086 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/f5jnqus6smazg)
- [elasticsearch-pull-request #151897 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/ukgbtgnh3n2rm)
- [elasticsearch-pull-request #151879 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/3rk3upnsqwp2e)
- [elasticsearch-pull-request #151851 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/mrq4odekhqbfk)
- [elasticsearch-periodic-java-ea #288 / release-tests](https://gradle-enterprise.elastic.co/s/ntuddb3v6dmw2)
- [elasticsearch-pull-request #151408 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/hbh6qo3lgbwrm)
- [elasticsearch-pull-request #151408 / checkpart2_release-tests](https://gradle-enterprise.elastic.co/s/lrygj6weqgqes)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:encryption:internalClusterTest" --tests "org.elasticsearch.xpack.encryption.KeyRotationIT.testInitialInstallProducesSingleActiveKeyWithEmptyHandlerProgress" -Dtests.seed=173991901A393E19 -Dtests.locale=vmw -Dtests.timezone=Asia/Khandyga -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.encryption.KeyRotationIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testInitialInstallProducesSingleActiveKeyWithEmptyHandlerProgress'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: install should produce exactly one key
Expected: a collection with size <1>
     but: collection size was <2>
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testInitialInstallProducesSingleActiveKeyWithEmptyHandlerProgress
- [main] 1 consecutive failures in step part2
- [main] 2 consecutive failures in step release-tests
- [main] 11 consecutive failures in step checkpart2_release-tests
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 14 failures in test testInitialInstallProducesSingleActiveKeyWithEmptyHandlerProgress (1.4% fail rate in 989 executions)
- [main] 2 failures in step release-tests (100.0% fail rate in 2 executions)
- [main] 11 failures in step checkpart2_release-tests (100.0% fail rate in 11 executions)
- [main] 9 failures in pipeline elasticsearch-pull-request (1.4% fail rate in 622 executions)
- [main] 2 failures in pipeline elasticsearch-periodic-java-ea (100.0% fail rate in 2 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -788,6 +788,3 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.ndjson.NdJsonCompressedFormatSpecIT
   method: test {csv-spec:external-basic.aggregateCount [ndjson.zstd/LOCAL]}
   issue: https://github.com/elastic/elasticsearch/issues/150723
-- class: org.elasticsearch.xpack.encryption.KeyRotationIT
-  method: testInitialInstallProducesSingleActiveKeyWithEmptyHandlerProgress
-  issue: https://github.com/elastic/elasticsearch/issues/150725

--- a/x-pack/plugin/encryption/src/internalClusterTest/java/org/elasticsearch/xpack/encryption/KeyRotationIT.java
+++ b/x-pack/plugin/encryption/src/internalClusterTest/java/org/elasticsearch/xpack/encryption/KeyRotationIT.java
@@ -18,6 +18,7 @@ import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.set.Sets;
 import org.elasticsearch.common.xcontent.ChunkedToXContentHelper;
 import org.elasticsearch.core.TimeValue;
 import org.elasticsearch.plugins.Plugin;
@@ -38,6 +39,8 @@ import java.util.EnumSet;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArrayList;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicInteger;
@@ -54,6 +57,8 @@ import static org.hamcrest.Matchers.notNullValue;
 
 @ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 3, supportsDedicatedMasters = false)
 public class KeyRotationIT extends SecurityIntegTestCase {
+
+    private final List<Integer> keyAddEvents = new CopyOnWriteArrayList<>();
 
     @Override
     protected boolean addMockHttpTransport() {
@@ -84,12 +89,30 @@ public class KeyRotationIT extends SecurityIntegTestCase {
     }
 
     @Before
-    public void waitForProjectEncryptionKeyInstalled() throws Exception {
+    public void setup() throws Exception {
+        // Check feature flag
         assumeTrue(
             "project encryption key feature flag must be enabled",
             ProjectEncryptionKeyService.PROJECT_ENCRYPTION_KEY_FEATURE_FLAG.isEnabled()
         );
+        // Add listener that registers PEK changes
+        internalCluster().clusterService().addListener(event -> {
+            if (event.changedCustomProjectMetadataSet().contains(ProjectEncryptionKeyMetadata.TYPE) == false) {
+                return;
+            }
+            ProjectEncryptionKeyMetadata curr = event.state().metadata().getSingleProjectCustom(ProjectEncryptionKeyMetadata.TYPE);
+            if (curr == null) {
+                return;
+            }
+            ProjectEncryptionKeyMetadata prev = event.previousState().metadata().getSingleProjectCustom(ProjectEncryptionKeyMetadata.TYPE);
+            Set<String> prevKeys = prev != null ? prev.getKeys().keySet() : Set.of();
+            var diff = Sets.difference(curr.getKeys().keySet(), prevKeys);
+            if (!diff.isEmpty()) {
+                keyAddEvents.add(diff.size());
+            }
+        });
         ensureGreen();
+        // Wait for first PEK install
         assertBusy(() -> assertThat(metadataOnMaster(), notNullValue()));
     }
 
@@ -159,12 +182,13 @@ public class KeyRotationIT extends SecurityIntegTestCase {
         }
     }
 
-    public void testInitialInstallProducesSingleActiveKeyWithEmptyHandlerProgress() {
-        ProjectEncryptionKeyMetadata metadata = metadataOnMaster();
-        assertThat(metadata, notNullValue());
-        assertThat("install should produce exactly one key", metadata.getKeys().keySet(), hasSize(1));
-        assertThat("the single installed key must be the active one", metadata.getKeys().keySet(), hasItem(metadata.getActiveKeyId()));
-        assertThat("no handlers are registered in this IT, so handlerKeyIds must be empty", metadata.getHandlerKeyIds(), anEmptyMap());
+    public void testEachInstallOrRotationAddsExactlyOneNewKey() throws Exception {
+        // Wait for two install or rotation events
+        assertBusy(() -> assertThat(keyAddEvents, hasSize(greaterThanOrEqualTo(2))), 30, TimeUnit.SECONDS);
+        for (int count : keyAddEvents) {
+            assertThat("each install or rotation must add exactly one key", count, equalTo(1));
+        }
+        assertThat("no handlers registered, handlerKeyIds must be empty", metadataOnMaster().getHandlerKeyIds(), anEmptyMap());
     }
 
     public void testRotationCycleWithNoHandlers() throws Exception {

